### PR TITLE
Remove AcidTest and bunch others

### PR DESCRIPTION
**Pull request by tomasbrod**
This fixes DoS attack reported by @barton2526 and performed by _tcblack_.
Also code 🔪

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -117,16 +117,7 @@ unsigned short GetListenPort()
...
-    std::string sboinchashargs = DefaultOrgKey(12);
-    std::string nonce = ToString(GetAdjustedTime());
-    std::string org = DefaultOrg();
-    std::string pub_key_prefix = OrgId();
-    std::string pw1 = RetrieveMd5(nonce+","+command+","+org+","+pub_key_prefix+","+sboinchashargs);
-    uint256 boincHashRandNonce = GetRandHash();
-    std::string bhrn = boincHashRandNonce.GetHex();
-    std::string grid_pass_encrypted = AdvancedCryptWithSalt(bhrn+nonce+org+pub_key_prefix,sboinchashargs);
-    std::string sComm = nonce+","+command+","+pw1+","+org+","+pub_key_prefix+","+bhrn+","+grid_pass_encrypted;
+    std::string sComm = "deprecated,d,d,d,d,d,d";
```
Comment (tomasbrod) [inline src/net.cpp]: This must have 7 items to avoid ban.

File: src/net.cpp [hunk 2]
```diff
@@ -741,18 +680,16 @@ void CNode::PushVersion()
...
-    uint256 boincHashRandNonce = GetRandHash();
-    std::string nonce = boincHashRandNonce.GetHex();
-    std::string pw1 = RetrieveMd5(nonce+","+sboinchashargs);
-    std::string mycpid = GlobalCPUMiningCPID.cpidv2;
+    std::string sboinchashargs = "";
+    std::string nonce = "";
+    std::string pw1 = "";
+    std::string mycpid = "INVESTOR";
     std::string acid = GetCommandNonce("aries");
-    std::string sGRCAddress = DefaultWalletAddress();
```
Comment (tomasbrod) [inline src/net.cpp]: Avoid leaking unnecessary info.

File: src/rpcblockchain.cpp [hunk 3]
```diff
@@ -2196,56 +2196,6 @@ Value execute(const Array& params, bool fHelp)
...
-            std::string sParam1 = params[1].get_str();
-            entry.push_back(Pair("Org",sParam1));
-            std::string sParam2 = params[2].get_str();
-            entry.push_back(Pair("Key",sParam2));
-            std::string key = sParam1 + "," + AdvancedDecryptWithSalt(sParam2,sParam1);
-            entry.push_back(Pair("PubKey",key));
-            results.push_back(entry);
-
-        }
-
```
Comment (tomasbrod) [inline src/rpcblockchain.cpp]: this was throwing errors so i deleted

File: src/main.cpp [hunk 4]
```diff
@@ -5944,48 +5944,7 @@ bool static AlreadyHave(CTxDB& txdb, const CInv& inv)
...
-        return true;
-    }
-    else
-    {
-        if (fDebug2) printf("Message corrupted. Node %s partially banned.",NodeAddress(pfrom).c_str());
-        pfrom->Misbehaving(1);
-        return false;
-    }
+    /* it did nothing anyway */
     return true;
```
Comment (tomasbrod) [inline src/main.cpp]: This check was crashing so skip that and just return true.

Comment (denravonska) [reply src/main.cpp]: We can remove the entire function and the checks that depend on it.

**Review by denravonska — CHANGES_REQUESTED** [review]
It would be nice if we could remove the dead functions so we don't have to be bothered by them anymore.

**Comment by tomasbrod** [comment]
@denravonska I made the suggested changes.
Before release, functionality of neural network must be verified.
More I noticed that one of my clients somehow connected two times to same node. There were two same addresses in getpeerinfo.

**Comment by iFoggz** [comment]
for out going connections u can only connect to one node ip in a certain size class range if i remember correctly. however if they have a ipv6 address as well u can then in turn have 2 connections to it (thou im not seeing this really commonly). incoming connections i believe don't have this rule. we should look into this more as well

**Pull request merged**